src/kbmod/trajectory_generator.py

```diff
@@ -314,25 +312,148 @@ def __init__(self, v_arr, ang_arr, average_angle, *args, **kwargs):
         ang_arr : `list`
             A triplet of the minimum angle offset (in radians), and the maximum angle offset
             (in radians), and the number of angles to try.
-        average_angle : `float`
+        average_angle : `float`, optional
             The central angle to search around. Should align with the ecliptic in most cases.
+        computed_ecliptic_angle : `float`, optional
+            An override for the computed ecliptic from a WCS (in the units defined in
+            ``angle_units``). This parameter is ignored if ``force_ecliptic`` is given.
         """
         if len(v_arr) != 3:
             raise ValueError("KBMODV1SearchConfig requires v_arr to be length 3")
         if len(ang_arr) != 3:
             raise ValueError("KBMODV1SearchConfig requires ang_arr to be length 3")
         if average_angle is None:
-            raise ValueError("KBMODV1SearchConfig requires a valid average_angle.")
+            if computed_ecliptic_angle is None:
+                raise ValueError(
+                    "KBMODV1SearchConfig requires a valid average_angle or computed_ecliptic_angle."
+                )
+                average_angle = computed_ecliptic_angle
 
         ang_min = average_angle - ang_arr[0]
         ang_max = average_angle + ang_arr[1]
-        super().__init__(v_arr[2], v_arr[0], v_arr[1], ang_arr[2], ang_min, ang_max, *args, **kwargs)
+        super().__init__(v_arr[2], v_arr[0], v_arr[1], ang_arr[2], ang_min, ang_max, **kwargs)
+
+
+class EclipticCenteredSearch(TrajectoryGenerator):
+    """Search a grid defined by velocities and angles relative to the ecliptic.
+
+    Attributes
+    ----------
+    ecliptic_angle : `float`
+        The angle to use for the ecliptic in the image (in the units defined in ``angle_units``).
+    velocities : `list`
+        A triplet of the minimum velocity to use (in pixels per day), and the maximum velocity
+        magnitude (in pixels per day), and the number of velocity steps to try.
+    angles : `list`
+        A triplet of the minimum angle offset (in radians), and the maximum angle offset (in radians),
+        and the number of angles to try.
+    min_ang : `float`
+        The minimum search angle in the image (ecliptic_angle + min_ang_offset) in radians.
+    max_ang : `float`
+        The maximum search angle in the image (ecliptic_angle + max_ang_offset) in radians.
+    """
+
+    def __init__(
+        self,
+        velocities=[0.0, 0.0, 0],
+        angles=[0.0, 0.0, 0],
+        angle_units="radians",
+        given_ecliptic=None,
+        computed_ecliptic_angle=None,
```

Comment:
Is there a point to differentiating between these two? I get having a flag in search config, but doesn't really seems necessary here.

Reply:
I think I have a better way to do this, which I will implement tomorrow.

Reply:
I changed the `TrajectoryGenerator` to take an optional reference to the `WorkUnit` so it can extract any information that it needs.